controllers/provider-aws/test/integration/healthcheck/test.go

```diff
@@ -0,0 +1,142 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+	Overview
+		- Tests the health checks of the extension: provider-aws.
+		- Manipulates health check relevant resources and expects the extension-provider to properly report the results as conditions in the respective CRD (ControlPlane(Type Normal & Exposure) & Worker CRD).
+
+	Prerequisites
+		- A Shoot exists.
+
+	Test-cases:
+		1) ControlPlane
+			1.1) HealthCondition Type: Shoot ControlPlaneHealthy
+				- delete the deployment 'cloud-controller-manager' and verify health check conditions in the ControlPlane status.
+			1.2) HealthCondition Type: Shoot SystemComponentsHealthy
+				- update the ManagedResource 'extension-controlplane-shoot' with an unhealthy condition and verify health check conditions in the ControlPlane status.
+		2) ControlPlane (Type: Exposure)
+			2.1) HealthCondition Type: Shoot ControlPlaneHealthy
+				- delete the deployment 'aws-lb-readvertiser' and verify health check conditions in the ControlPlane status.
+		3) Worker
+			3.1) HealthCondition Type: Shoot ControlPlaneHealthy
+				- delete the deployment 'machine-controller-manager' and verify health check conditions in the Worker status.
+			3.2) HealthCondition Type: Shoot SystemComponentsHealthy
+				- update the ManagedResource 'extension-worker-mcm-shoot' with an unhealthy condition and verify health check conditions in the Worker status.
+			3.3) HealthCondition Type: Shoot EveryNodeReady
+				- delete a machine of the shoot cluster and verify the health check conditions in the Worker status report a missing node.
+ **/
+
+package healthcheck
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/gardener/gardener-extensions/controllers/provider-aws/pkg/aws"
+	genericcontrolplaneactuator "github.com/gardener/gardener-extensions/pkg/controller/controlplane/genericactuator"
+	genericworkeractuator "github.com/gardener/gardener-extensions/pkg/controller/worker/genericactuator"
+	healthcheckoperation "github.com/gardener/gardener-extensions/test/integration/healthcheck"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	"github.com/gardener/gardener/test/framework"
+	machinev1alpha1 "github.com/gardener/machine-controller-manager/pkg/apis/machine/v1alpha1"
+	"github.com/onsi/ginkgo"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+)
+
+const (
+	timeout               = 10 * time.Minute
+	nodeRecreationTimeout = 20 * time.Minute
+	setupContextTimeout   = 2 * time.Minute
+)
+
+var _ = ginkgo.Describe("Provider-aws integration test: health checks", func() {
+	f := createShootFramework()
+
+	ginkgo.Context("Setup Check", func() {
+		f.Serial().CIt(fmt.Sprintf("Should be running on '%s'", aws.Type), func(ctx context.Context) {
```

Comment:
this test not do anything expect for printing out the provider.
I think we should check the shoots provider before every test and skip if the types do not match?

I suggest to add a `BeforeEach` that checks whether the shoot is applicable or not and skip the test if necessary.

Reply:
Why would you have that in a foreach? As far as I understand I am acting on the same shoot, retrieved by the Framework (that I created before the execution) - so only one preliminary check is necessary.
In the long-run I would like to see that kind of functionality in a common section for all providers e.g in the gardener - so that each extension-provider does not have to implement that check anymore.
WDYT

Reply:
I like the idea of this being part of the framework to verify such things there 👍 

Yes, you are only acting the one shoot created beforehand.
My point here is that this only gives an info that the tests are skipped (or should be skipped).
But it does not actually skip the tests if the shoot is of a different type, or am I missing something?
So for example if I would run this test on a GCP shoot, it would print the message that the provider is not aws but still execute all the tests.

Reply:
As discussed: removing provider checks as this repo is being split so that each provider has its own repository anyways.